File: Cliente1/ClienteManagement.vb

Public Class ClienteManagement

    Private crudCliente As AccesoDatos.Crud.ClienteCrudFactory

    Public Sub ClienteManagement()
        crudCliente = New AccesoDatos.Crud.ClienteCrudFactory()
    End Sub

    Public Function Create(cliente As Entidades.Cliente)
        crudCliente.Create(cliente)
    End Function

    Public Function RetrieveAll() As List(Of Entidades.Cliente)
        Return crudCliente.RetrieveAll(Of Entidades.Cliente)
    End Function

    Public Function RetrieveById(cliente As Entidades.Cliente) As Entidades.Cliente
        Return crudCliente.Retrieve(Of Entidades.Cliente)
    End Function

    Friend Function Update(cliente As Entidades.Cliente)
        crudCliente.Update(cliente)
    End Function

    Friend Function Delete(cliente As Entidades.Cliente)
        crudCliente.Delete(cliente)
    End Function

End Class

File: Management/ClienteManagement.cs
using AccesoDatos.Crud;
using Entidades;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Management
{
    public class ClienteManagement
    {
        private ClienteCrudFactory crudCliente;

        public ClienteManagement()
        {
            crudCliente = new ClienteCrudFactory();
        }

        public void Create(Cliente cliente)
        {

            crudCliente.Create(cliente);

        }


        public List<Cliente> RetrieveAll()
        {
            return crudCliente.RetrieveAll<Cliente>();
        }

        public Cliente RetrieveById(Cliente cliente)
        {
            return crudCliente.Retrieve<Cliente>(cliente);
        }

        public void Update(Cliente cliente)
        {
            crudCliente.Update(cliente);
        }

        public void Delete(Cliente cliente)
        {
            crudCliente.Delete(cliente);
        }
    }
}
